src-tauri/src/lib/utils.rs

```diff
@@ -19,8 +19,40 @@ use std::{
     fs::{self, File},
     io::{self, BufReader, Read},
     path::{Path, PathBuf},
+    time::{Duration, Instant},
+    cmp::Ordering,
 };
 use regex::Regex;
+use url::Url;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MirrorEntry {
+    pub url: String,
+    pub latency: Option<u32>,
+}
+
+impl PartialOrd for MirrorEntry {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for MirrorEntry {
+    fn cmp(&self, other: &Self) -> Ordering {
```

Comment:
Much better approach! But still ... Option<T> already implements the Ord Trait. So it can be simplified like this:
```
match (self.latency, other.latency) {
    (Some(a), Some(b)) => a.cmp(&b),
    (Some(_), None) => Ordering::Less,
    (None, Some(_)) => Ordering::Greater,
    (None, None) => Ordering::Equal,
}
```
This 1) avoids unwrap() 2) ensures all possible cases are covered (match will not let you have unimplemented variants) 3) is clearly readable 4) pattern matching is the rust way of doing things ;-)